### PR TITLE
Cleanup hooks

### PR DESCRIPTION
**Pull request by schiessle**
The group manager Hooks are quite a mess, some are called for the class OC_User and some for the class OC_Group.

This is a first suggestion how to clean it up, as discussed with @blizzz

### CODE REVIEW

**Comment by schiesbn** [comment]
@blizzz what do you think? now every hook is emitted by the class oc_group, expect the addUser hooks. Maybe for consistency reasons it would make sense to emit them also by the class oc_group. At the end it is a group action triggered by the GroupManager.... Opinions?

**Comment by DeepDiver1975** [comment]
> Opinions?

kill the hooks .... use the listener mechanism (if applies)

**Comment by blizzz** [comment]
@DeepDiver1975 LDAP  backend "misuses" the hooks to announce changed group memberships. As far as i can see this is no longer possible with the Listener approach as the instance has a protected emit(). I.e. we'd need to drill another hole somewhere in  the user management to  allow backends to throw such announcements. 

Of course, this is more of a cleaner approach than the old hack…

**Comment by schiesbn** [comment]
@owncloud-bot retest this please

**Comment by rullzer** [comment]
@blizzz not really drill a hole... just add more listners and hooks so the backend can properly announce it.

**Comment by PVince81** [comment]
@blizzz @rullzer @schiesbn what to do here ?

**Comment by MorrisJobke** [comment]
@blizzz @rullzer @schiesbn An update on this would be nice - even if it is an "move to 9.0" ;)

**Comment by rullzer** [comment]
I guess move to 9.0 since breaking LDAP for 8.2 is not really an option

**Comment by karlitschek** [comment]
yep. too late for 8.2

**Comment by PVince81** [comment]
We could keep both for now (the old one and the new one) a deprecate the old one until the LDAP code could migrate ? Not sure, but if that doesn't happen it means even more clutter.

**Comment by LukasReschke** [comment]
@schiesbn Has conflicts. Can you please rebase? THX.

File: lib/base.php [hunk 1]
```diff
@@ -785,10 +785,10 @@ public static function registerPreviewHooks() {
 	public static function registerShareHooks() {
 		if (\OC::$server->getSystemConfig()->getValue('installed')) {
 			OC_Hook::connect('OC_User', 'post_deleteUser', 'OC\Share\Hooks', 'post_deleteUser');
-			OC_Hook::connect('OC_User', 'post_addToGroup', 'OC\Share\Hooks', 'post_addToGroup');
+			OC_Hook::connect('OC_Group', 'post_addToGroup', 'OC\Share\Hooks', 'post_addToGroup');
```
Comment (MorrisJobke) [inline lib/base.php]: This is the only critical merge conflict - @schiesbn you changed this in https://github.com/owncloud/core/commit/058d910f5e85ca71d20272d8ca89dc3caf64cef2 already. Is the above change then still needed?

**Comment by MorrisJobke** [comment]
> @schiesbn Has conflicts. Can you please rebase? THX.

Done. I commented on the line, where I'm not sure how it should work - @schiesbn needs to comment there.

**Comment by PVince81** [comment]
Risky change ? Move to 9.1 ?

**Comment by PVince81** [comment]
Tech debt, Moving to 9.1

**Comment by PVince81** [comment]
This PR still looks risky, I suggest closing for now and reconsider this the next time a task comes up that involves those hooks, maybe an opportunity to move these to the event dispatcher pattern.

@schiesbn what do you think ?

**Pull request closed without merging**